pumpkin-protocol/src/query.rs

```diff
@@ -0,0 +1,89 @@
+use tokio::io::AsyncReadExt;
+
+#[derive(Debug)]
+pub enum PacketType {
+    Handshake = 9,
+    Stat = 0,
+}
+
+#[derive(Debug)]
+pub struct SBasePacket {
+    pub magic: u16,
+    pub packet_type: PacketType,
```

Comment:
In the docs it says that the packet type is a byte. You can still use the Enum but you should add `repr(i8)`